frontend/packages/dev-console/src/components/ProjectSelectPage.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { match } from 'react-router';
+import { LoadingBox } from '@console/internal/components/utils';
+import { connectToPlural } from '@console/internal/kinds';
+import {
+  apiVersionForReference,
+  isGroupVersionKind,
+  kindForReference,
+} from '@console/internal/module/k8s';
+import { ErrorPage404 } from '@console/internal/components/error';
+import { withStartGuide } from '../../../../public/components/start-guide';
+import ProjectListPage from './projects/ProjectListPage';
+
+export interface ProjectSelectPageProps {
+  match: match<any>;
+}
+
+const allParams = (props) => Object.assign({}, props?.match?.params, props);
+
+const ProjectSelectPage: React.FC<ProjectSelectPageProps> = (props) => {
+  const { kindObj, kindsInFlight, plural } = allParams(props);
+
+  if (!kindObj) {
+    if (kindsInFlight) {
+      return <LoadingBox />;
+    }
+    const missingType = isGroupVersionKind(plural)
+      ? `"${kindForReference(plural)}" in "${apiVersionForReference(plural)}"`
+      : `"${plural}"`;
+    return (
+      <ErrorPage404
+        message={`The server doesn't have a resource type ${missingType}. Try refreshing the page if it was recently added.`}
+      />
+    );
+  }
+  return (
+    <>
+      <Helmet>
+        <title>{kindObj.labelPlural}</title>
+      </Helmet>
+      <ProjectListPage title={kindObj.labelPlural}>
+        {`Select a project to view the list of ${kindObj.labelPlural}`}
+      </ProjectListPage>
+    </>
+  );
+};
+
+export default connectToPlural(withStartGuide(ProjectSelectPage));
```

Comment:
`connectToPlural` is deprecated. Let me see what an alternative would be.

Reply:
The only alternative I can see is if you thread the `model` from `getPinnedItems` (in [perspective-nav.tsx](https://github.com/openshift/console/blob/master/frontend/public/components/nav/perspective-nav.tsx)) and use React-Router-Dom `to` as an [object](https://reacttraining.com/react-router/web/api/Link/to-object) and provide the state.

Reply:
I think its best to leave as is and someday when the connectToPlural is really removed it is replaced with a viable alternative